pkg/smi/access/v1alpha2/methods.go

```diff
@@ -34,3 +39,23 @@ func doesTargetRefSrcPod(spec accessClient.TrafficTargetSpec, srcPod *corev1.Pod
 	}
 	return false
 }
+
+// GetExistingRouteNames returns the names of HTTPRouteGroups and TCPRoutes that exist in the cluster
+func GetExistingRouteNames(specClient smiSpecClient.Interface, namespace string) (mapset.Set, error) {
+	routes := mapset.NewSet()
+	httpRouteGroups, err := specClient.SpecsV1alpha3().HTTPRouteGroups(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msgf("Error getting HTTPRouteGroups for namespace %s", namespace)
+	}
+	for _, httpRouteGroup := range httpRouteGroups.Items {
+		routes.Add(httpRouteGroup.Name)
+	}
+	tcpRoutes, err := specClient.SpecsV1alpha3().TCPRoutes(namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		log.Err(err).Msgf("Error getting HTTPRouteGroups for namespace %s", namespace)
+	}
+	for _, tcpRoute := range tcpRoutes.Items {
+		routes.Add(tcpRoute.Name)
+	}
+	return routes, err
```

Comment:
nit: If an err occurs for both `List` calls, the first `err` will be overwritten and only the second `err` will be returned. Most likely both errors would be the same. I don't think this needs to be addressed.

Reply:
Yeah that's what I was thinking!